torch_xla/core/xla_model.py

```diff
@@ -553,7 +553,7 @@ def all_gather(value, dim=0, groups=None, output=None, pin_layout=True):
     A tensor which has, in the ``dim`` dimension, all the values from the
     participating replicas.
   """
-  if pin_layout and (output == None or xla_device_hw(value.device) == 'NEURON'):
+  if pin_layout and output == None and isinstance(value, torch.Tensor):
```

Comment:
Could you add a comment to explain the check here? I think it means we can't use `_all_gather_using_all_reduce` if input is a list or `output` is not none? I forgot why we added `xla_device_hw(value.device) == 'NEURON'` at first place..

Reply:
Will add a note. Yes, _all_gather_using_all_reduce doesn't work with coaelescing (multiple tensor list).